apps/server/src/infra/sync/tsp/tsp-legacy-migration.service.ts

```diff
@@ -0,0 +1,93 @@
+import { EntityManager, ObjectId } from '@mikro-orm/mongodb';
+import { Injectable } from '@nestjs/common';
+import { EntityId, SchoolFeature } from '@shared/domain/types';
+import { Logger } from '@src/core/logger';
+import { TspLegacyMigrationStartLoggable } from './loggable/tsp-legacy-migration-start.loggable';
+import { TspLegacyMigrationSystemMissingLoggable } from './loggable/tsp-legacy-migration-system-missing.loggable';
+import { TspLegacySchoolMigrationCountLoggable } from './loggable/tsp-legacy-school-migration-count.loggable';
+import { TspLegacySchoolMigrationSuccessLoggable } from './loggable/tsp-legacy-school-migration-success.loggable';
+
+type LegacyTspSchoolProperties = {
+	sourceOptions: {
+		schoolIdentifier: number;
+	};
+};
+
+const TSP_LEGACY_SYSTEM_TYPE = 'tsp-school';
+const TSP_LEGACY_SOURCE_TYPE = 'tsp';
+const SCHOOLS_COLLECTION = 'schools';
+const SYSTEMS_COLLECTION = 'systems';
+
+@Injectable()
+export class TspLegacyMigrationService {
+	constructor(private readonly em: EntityManager, private readonly logger: Logger) {
+		logger.setContext(TspLegacyMigrationService.name);
+	}
+
+	public async migrateLegacyData(newSystemId: EntityId): Promise<void> {
+		this.logger.info(new TspLegacyMigrationStartLoggable());
+
+		const legacySystemId = await this.findLegacySystemId();
+
+		if (!legacySystemId) {
+			this.logger.info(new TspLegacyMigrationSystemMissingLoggable());
+			return;
+		}
+
+		const schoolIds = await this.findIdsOfLegacyTspSchools(legacySystemId);
+
+		this.logger.info(new TspLegacySchoolMigrationCountLoggable(schoolIds.length));
+
+		const promises = schoolIds.map(async (oldId): Promise<number> => {
+			const legacySchoolFilter = {
+				systems: [legacySystemId],
+				source: TSP_LEGACY_SOURCE_TYPE,
+				sourceOptions: {
+					schoolIdentifier: oldId,
+				},
+			};
+
+			const featureUpdateCount = await this.em.nativeUpdate(SCHOOLS_COLLECTION, legacySchoolFilter, {
+				$addToSet: {
+					features: SchoolFeature.OAUTH_PROVISIONING_ENABLED,
+				},
+			});
+			const idUpdateCount = await this.em.nativeUpdate(SCHOOLS_COLLECTION, legacySchoolFilter, {
+				ldapSchoolIdentifier: oldId,
+				systems: [new ObjectId(newSystemId)],
+			});
+
+			return featureUpdateCount === 1 && idUpdateCount === 1 ? 1 : 0;
```

Comment:
missing test case after ?

Reply:
@MajedAlaitwniCap It would be very time-consuming to test this. And since Sonar doesn't complain and the code is thrown away anyway, we would skip the test.